src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -1 +1,116 @@
+    Query query = new Query("Comments").addSort("score-class", SortDirection.ASCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      String comment = (String) entity.getProperty("comments-class");
+      json.add(comment);
+    }
+
+    // Send JSON string.
+    String jsonVersion = new Gson().toJson(json);
+    response.setContentType("application/json;");
+    response.getWriter().println(jsonVersion);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get written feedback.
+    String classFeedback = request.getParameter("class-input");
+    String professorFeedback = request.getParameter("prof-input");
+    // Get class/professor ratings.
+    String ratingClass = request.getParameter("rating-class");
+    String ratingProfessor = request.getParameter("rating-professor");
+    boolean translateEnglish = Boolean.parseBoolean(request.getParameter("languages"));
+
+    if (translateEnglish) {
+      Translate translate = TranslateOptions.getDefaultInstance().getService();
+      Translation translationClassFeedback =
+          translate.translate(classFeedback, Translate.TranslateOption.targetLanguage("en"));
+      Translation translationProfessorFeedback =
+          translate.translate(professorFeedback, Translate.TranslateOption.targetLanguage("en"));
+      String translatedClassFeedback = translationClassFeedback.getTranslatedText();
+      String translatedProfessorFeedback = translationProfessorFeedback.getTranslatedText();
+      classFeedback = translatedClassFeedback;
+      professorFeedback = translatedProfessorFeedback;
+    }
+
+    Document docClass =
+        Document.newBuilder().setContent(classFeedback).setType(Document.Type.PLAIN_TEXT).build();
+    Document docProfessor =
+        Document.newBuilder()
+            .setContent(professorFeedback)
+            .setType(Document.Type.PLAIN_TEXT)
+            .build();
+
+    LanguageServiceClient languageServiceForClass = LanguageServiceClient.create();
+    Sentiment sentimentForClass =
+        languageServiceForClass.analyzeSentiment(docClass).getDocumentSentiment();
+    float scoreClass = sentimentForClass.getScore();
+    languageServiceForClass.close();
+
+    LanguageServiceClient languageServiceForProf = LanguageServiceClient.create();
+    Sentiment sentimentForProf =
+        languageServiceForProf.analyzeSentiment(docProfessor).getDocumentSentiment();
+    float scoreProfessor = sentimentForProf.getScore();
+    languageServiceForProf.close();
+
+    Entity professorEntity = new Entity("Professor");
+    Entity commentsEntity = new Entity("Comments");
+    professorEntity.setProperty("comments-professor", professorFeedback);
+    commentsEntity.setProperty("comments-class", classFeedback);
```

Comment:
More of a nit could you reorganize this section so you add all the commentsEntity properties then the professorEntity